### PR TITLE
better handling of preview generation errors

### PR DESCRIPTION
**Pull request by icewind1991**
Prevents failing hard when working with corrupted images

### CODE REVIEW

**Comment by MorrisJobke** [comment]
Makes sense but I could not verify that it failed before. I took a document and renamed it to .png and triggered the occ preview:generate-all - before and after there is no error.

**Comment by icewind1991** [comment]
You need an actual corrupted image, not just a wrongly named one

**Review by rullzer — APPROVED** [review]
Makes sense!

**Comment by MorrisJobke** [comment]
> You need an actual corrupted image, not just a wrongly named one

Do you have one?

**Comment by MorrisJobke** [comment]
I also added some error handling for the other endpoints that uses that method.

**Review by MorrisJobke — APPROVED** [review]

File: lib/private/Preview/Generator.php [hunk 1]
```diff
@@ -325,6 +327,10 @@ private function generatePreview(ISimpleFolder $previewFolder, ISimpleFile $maxP
 			$preview->resize(max($width, $height));
 		}
 
+		if (!$preview->valid()) {
```
Comment (rullzer) [inline lib/private/Preview/Generator.php]: Shouldn't this be moved up?  Like right after the getImage call?

**Review by rullzer — CHANGES_REQUESTED** [review]

File: lib/private/Preview/Generator.php (hunk 1 above)
Comment (icewind1991) [inline lib/private/Preview/Generator.php]: done

**Review by rullzer — APPROVED** [review]

**Pull request merged**